qiskit_nature/properties/second_quantization/electronic/angular_momentum.py

```diff
@@ -58,14 +65,25 @@ def from_driver_result(cls, result: DriverResult) -> "AngularMomentum":
         Raises:
             QiskitNatureError: if a WatsonHamiltonian is provided.
         """
-        cls._validate_input_type(result, ElectronicDriverResult)
+        cls._validate_input_type(result, LegacyElectronicDriverResult)
 
         qmol = cast(QMolecule, result)
 
         return cls(
             qmol.num_molecular_orbitals * 2,
         )
 
+    def reduce_system_size(self, active_orbital_indices: List[int]) -> "AngularMomentum":
```

Comment:
Yes I am also not too happy with this yet. At first I thought that (since size reduction is so important) it is fine if we make this more of a first-class citizen in the `Property` objects. But I agree now that this is bad design and should purely be handled by the transformer. I will update that later.